ceph: operator is kiled due to SIGSEGV in OBC creation

**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**

Creating OBC wasn't finished . After that, operator is killed in ObjectStore reconciliation due to SEGSEGV.

The log in OBC creation.

```
E0219 13:06:42.451901       8 controller.go:199] error syncing 'logging/loki-bucket': error provisioning bucket: Provision: can't create ceph user: no user name provided and unable to generate a unique name: error getting ceph user "ceph-user-pbD3uYEg": radosgw-admin command err. 2021-02-19T13:06:42.405+0000 7f922a82c980  0 failed reading zonegroup info: ret -2 (2) No such file or directory
2021-02-19T13:06:42.405+0000 7f922a82c980  0 ERROR: failed to start notify service ((2) No such file or directory
2021-02-19T13:06:42.405+0000 7f922a82c980  0 ERROR: failed to init services (ret=(2) No such file or directory)
couldn't init storage provider: exit status 5, requeuing
2021-02-19 13:06:58.459911 I | cephclient: creating replicated pool ceph-hdd-object-store.rgw.log succeeded
2021-02-19 13:06:58.459995 I | cephclient: setting pool property "pg_num_min" to "8" on pool "ceph-hdd-object-store.rgw.log"
2021-02-19 13:07:22.950476 I | cephclient: creating replicated pool ceph-hdd-object-store.rgw.buckets.index succeeded
2021-02-19 13:07:22.950600 I | cephclient: setting pool property "pg_num_min" to "8" on pool "ceph-hdd-object-store.rgw.buckets.index"
2021-02-19 13:07:44.747918 I | cephclient: creating replicated pool ceph-hdd-object-store.rgw.buckets.non-ec succeeded
2021-02-19 13:07:44.748001 I | cephclient: setting pool property "pg_num_min" to "8" on pool "ceph-hdd-object-store.rgw.buckets.non-ec"
2021-02-19 13:07:50.374674 I | cephclient: setting pool property "pg_num_min" to "8" on pool ".rgw.root"
2021-02-19 13:08:14.834990 I | cephclient: creating replicated pool ceph-hdd-object-store.rgw.buckets.data succeeded
2021-02-19 13:08:14.835205 I | ceph-object-controller: setting multisite settings for object store "ceph-hdd-object-store"
2021-02-19 13:08:34.697009 I | ceph-object-controller: Multisite for object-store: realm=ceph-hdd-object-store, zonegroup=ceph-hdd-object-store, zone=ceph-hdd-object-store
2021-02-19 13:08:34.697155 I | ceph-object-controller: multisite configuration for object-store ceph-hdd-object-store is complete
2021-02-19 13:08:34.697182 I | ceph-object-controller: creating object store "ceph-hdd-object-store" in namespace "ceph-hdd"
2021-02-19 13:08:34.697261 I | cephclient: getting or creating ceph auth key "client.rgw.ceph.hdd.object.store.a"
2021-02-19 13:08:37.559798 I | ceph-object-controller: setting rgw config flags
2021-02-19 13:08:54.043187 I | ceph-object-controller: object store "ceph-hdd-object-store" deployment "rook-ceph-rgw-ceph-hdd-object-store-a" started
2021-02-19 13:08:57.825372 I | ceph-object-controller: enabling rgw dashboard
2021-02-19 13:08:59.015798 I | ceph-object-controller: created object store "ceph-hdd-object-store" in namespace "ceph-hdd"
2021-02-19 13:08:59.015878 I | ceph-object-controller: starting rgw healthcheck
2021-02-19 13:09:14.018946 I | exec: timeout waiting for process radosgw-admin to return. Sending interrupt signal to the process
2021-02-19 13:10:29.510152 I | exec: timeout waiting for process radosgw-admin to return. Sending interrupt signal to the process
2021-02-19 13:11:46.071351 I | exec: timeout waiting for process radosgw-admin to return. Sending interrupt signal to the process
2021-02-19 13:13:02.489702 I | exec: timeout waiting for process radosgw-admin to return. Sending interrupt signal to the process
2021-02-19 13:14:18.314422 I | exec: timeout waiting for process radosgw-admin to return. Sending interrupt signal to the process
2021-02-19 13:15:33.728969 I | exec: timeout waiting for process radosgw-admin to return. Sending interrupt signal to the process
```

After showing above logs, operator was killed with the following log.

```
kubectl -n ceph-hdd logs rook-ceph-operator-85d5bf44b-6s8nx -p | tail
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1ddbc59]

goroutine 1966 [running]:
github.com/rook/rook/pkg/operator/ceph/object.(*bucketChecker).checkObjectStoreHealth(0xc0012dd880, 0x15, 0xc001006770)
        /work/go/src/github.com/rook/rook/pkg/operator/ceph/object/health.go:141 +0x9f9
github.com/rook/rook/pkg/operator/ceph/object.(*bucketChecker).checkObjectStore(0xc0012dd880, 0xc000bc2960)
        /work/go/src/github.com/rook/rook/pkg/operator/ceph/object/health.go:99 +0x17b
created by github.com/rook/rook/pkg/operator/ceph/object.(*ReconcileCephObjectStore).startMonitoring
        /work/go/src/github.com/rook/rook/pkg/operator/ceph/object/controller.go:514 +0x33b
```

In addition, after rebooting operator, it continued to fail with the following log.

```
2021-02-19 13:21:12.832216 I | op-bucket-prov: Provision: creating bucket "loki-bucket-5f44c382-03f6-4435-a28c-d798c9d2af7f" for OBC "loki-bucket"
2021-02-19 13:21:27.844268 I | exec: timeout waiting for process radosgw-admin to return. Sending interrupt signal to the process
E0219 13:21:27.851358       8 controller.go:199] error syncing 'logging/loki-bucket': error provisioning bucket: Provision: can't create ceph user: no user name provided and unable to generate a unique name: error getting ceph user "ceph-user-SsLR0b90": radosgw-admin command err. : signal: interrupt, requeuing
I0219 13:21:32.971904       8 controller.go:212]  "msg"="reconciling claim" "key"="logging/loki-bucket" 
I0219 13:21:32.972567       8 helpers.go:107]  "msg"="getting claim for key" "key"="logging/loki-bucket" 
I0219 13:21:32.992479       8 helpers.go:213]  "msg"="getting ObjectBucketClaim's StorageClass" "key"="logging/loki-bucket" 
I0219 13:21:34.229597       8 helpers.go:218]  "msg"="got StorageClass" "key"="logging/loki-bucket" "name"="ceph-hdd-bucket"
I0219 13:21:34.229827       8 helpers.go:90]  "msg"="checking OBC for OB name, this indicates provisioning is complete" "key"="logging/loki-bucket" "loki-bucket"=null
I0219 13:21:34.230218       8 resourcehandlers.go:446]  "msg"="updating status:" "key"="logging/loki-bucket" "new status"="Pending" "obc"="logging/loki-bucket" "old status"="Pending"
I0219 13:21:34.786450       8 controller.go:273]  "msg"="syncing obc creation" "key"="logging/loki-bucket" 
I0219 13:21:34.786678       8 controller.go:620]  "msg"="updating OBC metadata" "key"="logging/loki-bucket" 
I0219 13:21:34.786701       8 resourcehandlers.go:436]  "msg"="updating" "key"="logging/loki-bucket" "obc"="logging/loki-bucket"
I0219 13:21:35.483637       8 resourcehandlers.go:148]  "msg"="seeing if OB for OBC exists" "key"="logging/loki-bucket" "checking for OB name"="obc-logging-loki-bucket"
I0219 13:21:35.871583       8 controller.go:396]  "msg"="provisioning" "key"="logging/loki-bucket" "bucket"="loki-bucket-5f44c382-03f6-4435-a28c-d798c9d2af7f"
```

Both restarting rgw and mgr don't change the above-mentione continuous error.

OBC binding succedded when deleting it and removing finalizers.

If succeeded to create user, the log is as follows.

```
2021-02-19 13:41:07.387351 I | op-bucket-prov: Provision: creating bucket "loki-bucket-60521fe6-c8d1-4b41-9872-c38bca81829e" for OBC "loki-bucket"
2021-02-19 13:41:17.975302 I | op-bucket-prov: creating Ceph user "ceph-user-7qJ6zIzi"
2021-02-19 13:41:30.013608 I | op-bucket-prov: successfully created Ceph user "ceph-user-7qJ6zIzi" with access keys
```

However, in my case, operator continued to timeout (15s) in user creating.

``` 
2021-02-19 13:39:30.046642 I | op-bucket-prov: Provision: creating bucket "loki-bucket-60521fe6-c8d1-4b41-9872-c38bca81829e" for OBC "loki-bucket"
2021-02-19 13:39:45.049624 I | exec: timeout waiting for process radosgw-admin to return. Sending interrupt signal to the process
E0219 13:39:45.059698       8 controller.go:199] error syncing 'logging/loki-bucket': error provisioning bucket: Provision: can't create ceph user: no user name provided and unable to generate a unique name: error getting ceph user "ceph-user-deucliC3": radosgw-admin command err. : signal: interrupt, requeuing
```

**Expected behavior:**

OBC works fine.

**How to reproduce it (minimal and precise):**

Not sure yet.

**File(s) to submit**:

* Cluster CR (custom resource), typically called `cluster.yaml`, if necessary
* Operator's logs, if necessary
`
``

```

* Crashing pod(s) logs, if necessary

 To get logs, use `kubectl -n <namespace> logs <pod name>`
When pasting logs, always surround them with backticks or use the `insert code` button from the Github UI.
Read [Github documentation if you need help](https://help.github.com/en/articles/creating-and-highlighting-code-blocks).

**Environment**:
* OS (e.g. from /etc/os-release):
* Kernel (e.g. `uname -a`):
* Cloud provider or hardware configuration:
* Rook version (use `rook version` inside of a Rook Pod): v1.5.7
* Storage backend version (e.g. for ceph do `ceph -v`): ceph
* Kubernetes version (use `kubectl version`): v1.18
* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift):
* Storage backend status (e.g. for Ceph use `ceph health` in the [Rook Ceph toolbox](https://rook.io/docs/rook/master/ceph-toolbox.html)):

--- a/pkg/operator/ceph/object/health.go
+++ b/pkg/operator/ceph/object/health.go
@@ -138,7 +138,7 @@ func (c *bucketChecker) checkObjectStoreHealth() error {
 		if rgwerr == ErrorCodeFileExists {
 			user, _, err = GetUser(c.objContext, userConfig.UserID)
 			if err != nil {
-				return errors.Wrapf(err, "failed to get details from ceph object user %q for object store %q", user.UserID, c.namespacedName.Name)
+				return errors.Wrapf(err, "failed to get details from ceph object user %q for object store %q", userConfig.UserID, c.namespacedName.Name)
 			}
 		} else {
 			return errors.Wrapf(err, "failed to create object user %q. error code %d for object store %q", userConfig.UserID, rgwerr, c.namespacedName.Name)

